picard/tagger.py

```diff
@@ -745,10 +745,9 @@ def use_acoustid(self):
     def analyze(self, objs):
         """Analyze the file(s)."""
         files = self.get_files_from_objects(objs)
+        use_acoustid = self.use_acoustid
```

Comment:
This looks to me a bit like over-optimization, could just as well use `self.use_acoustid` at the one line it is used in the loop.

Reply:
Last time i measured performance regarding use of `config` i was surprised how slow it was, so i tend to avoid it in loops etc... i could spend time to measure this again though.

But i disagree on "over-optimization" here, the fact is this code is very "under-optimized" by design, doing an invariant test in a loop and hiding the complexity behind fake properties.

That's one drawback of Python "features": `self.use_acoustid` is not a variable but a method doing a test calling `config` which is itself a bunch of methods with a disk backend... 

Though i agree resulting code is "weird", i'd rather write it:
`use_acoustid = self.use_tons_of_stuff_you_d_better_cache()` to start with...

Reply:
I removed changes concerning acoustid stuff from this PR

Reply:
Ok, but you made valid points here. Would be an argument making this a function instead of a property as you suggest.